File: app/src/main/java/app/felgueiras/musicapp/api/Artist.java
package app.felgueiras.musicapp.api;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

import java.io.Serializable;
import java.util.List;

/**
 * Created by felguiras on 27/09/2017.
 */

@Root(name = "artist", strict = false)
public class Artist implements Serializable {

    @Element(name = "name")
    public String name;

    @Element(name = "mbid")
    public String mbid;

    @Element(name = "image")
    public List<ImageInfo> image;

//    public String getmChannel() {
//        return name;
//    }

    public Artist() {
    }

    @Root(name = "image", strict = false)
    public class ImageInfo implements Serializable{

        // TODO - access this field
        @Element(name = "#text")
        public String text;

        @Element(name = "size")
        public String size;

        public ImageInfo() {
        }
    }

//    public Tracks(String mChannel) {
//        this.name = mChannel;
//    }
//
//    public String getName() {
//        return name;
//    }
//
//    public void setName(String name) {
//        this.name = name;
//    }
}

File: app/src/main/java/app/felgueiras/musicapp/api/Artist.kt
package app.felgueiras.musicapp.api


import com.google.gson.annotations.SerializedName
import java.io.Serializable


/**
 * Artist POJO.
 */
class Artist : Serializable {

    override fun toString(): String {
        return "Artist: ${name!!}\nBio: ${bio.summary}"
    }

    /**
     * Name.
     */
    var name: String? = null

    /**
     * Unique ID
     */
    lateinit var mbid: String

    /**
     * Images (different sizes)
     */
    @SerializedName("image")
    lateinit var images: List<ImageInfo>


    /**
     * Similar artists.
     */
    lateinit var similar: Similar

    /**
     * Tags (genres)
     */
    lateinit var tags: Tags

    /**
     * Biography.
     */
    lateinit var bio: Bio

    inner class Similar : Serializable {

        lateinit var artist: List<Artist>
    }

    inner class Bio : Serializable {

        /**
         * Resumed bio.
         */
        lateinit var summary: String

        /**
         * Full bio.
         */
        lateinit var content: String
    }

    inner class Tags : Serializable {

        lateinit var tag: List<Tag>
    }

    inner class Tag : Serializable {

        /**
         * Genre.
         */
        lateinit var name: String
    }

}
